src/SourceIndex/index.proj

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <ItemGroup>
+    <SourceProject Include="../**/src/*.csproj"/>
+    <SourceProject Include="../**/tests/*.csproj"/>
+    <SourceProject Include="../**/src/*.vbproj"/>
+    <SourceProject Include="../**/tests/*.vbproj"/>
+    <SourceProject Include="../../../coreclr/src/mscorlib/System.Private.CoreLib.csproj"/>
+  </ItemGroup>
+
+  <UsingTask TaskName="SortItems" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v4.0.dll">
+    <ParameterGroup>
+      <Inputs ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true"/>
+      <Outputs ParameterType="Microsoft.Build.Framework.ITaskItem[]" Output="true"/>
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System"/>
+      <Using Namespace="System.Linq"/>
+      <Using Namespace="Microsoft.Build.Framework"/>
+      <Code>
+      <![CDATA[
+      Outputs = Inputs.OrderBy(i => i.GetMetadata("Identity")).ToArray();
+      ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+  <Target Name="RestorePackages" Condition="'$(HtmlGeneratorExePath)' == ''">
+    <Exec Command="$(DotnetToolCommand) restore --packages $(PackagesDir)" WorkingDirectory="$(MSBuildThisFileDirectory)"/>
+    <PropertyGroup>
+      <HtmlGeneratorPath>$(PackagesDir)Microsoft.SourceBrowser/1.0.13/tools/</HtmlGeneratorPath>
+      <HtmlGeneratorExePath>$(HtmlGeneratorPath)HtmlGenerator.exe</HtmlGeneratorExePath>
+    </PropertyGroup>
+    <Copy SourceFiles="$(MSBuildThisFileDirectory)HtmlGenerator.exe.config" DestinationFiles="$(HtmlGeneratorPath)HtmlGenerator.exe.config"/>
+  </Target>
+
+  <Target Name="Build" DependsOnTargets="RestorePackages">
```

Comment:
I'm guessing this would only work on Windows, as the tool seems to be a .NET Framework program. Is it runnable with mono, or should we handle that case in some other way (maybe just error out with a better error)?